### PR TITLE
Bugfixes and implemented an error catch block to utils func

### PR DESCRIPTION
**Pull request by Keilvin**

### CODE REVIEW

File: resources/mentors.py [hunk 1]
```diff
@@ -184,15 +185,16 @@ def post(self):
                 conn.close()
 
 class GetMentorQuestions(Resource):
+    @jwt_required
 
     # @jwt_required
-    def post(self):
```
Comment (JustinRMoy) [inline resources/mentors.py:189]: this needs to be a post and NOT a get. This will ruin Unity's ability to interact with the server

File: resources/mentors.py [hunk 2]
```diff
@@ -319,14 +321,14 @@ def delete(self):
 
 
 class GetMultipleChoiceOptions(Resource):
-    # @jwt_required
-    def post(self):
```
Comment (JustinRMoy) [inline resources/mentors.py:323]: same here, these were split off and made posts for the reason stated above

File: utils.py [hunk 3]
```diff
@@ -389,15 +389,24 @@ def getTimeDiffFormatted(time_1 = None,
...
         time_delta = time_obj
     else:
         return None, None
-    if time_delta.days != 0:  
+
+    if time_delta.days != 0:
         d = {"days": time_delta.days}
-        d["hours"], rem = divmod(time_delta.seconds, 3600)
+        hours, rem = divmod(time_delta.seconds, 3600)
+        d["hours"] = hours
```
Comment (JustinRMoy) [inline utils.py]: what's the point of this change?

**Review by JustinRMoy — COMMENTED** [review]
Undo the changes of making Post requests Get. Adress the other comments

File: resources/group.py [hunk 4]
```diff
@@ -26,7 +26,7 @@ def post(self):
             conn = mysql.connect()
             cursor = conn.cursor()
 
-            if permission != 'pf':
+            if permission != 'pf' or permission != 'su':
```
Comment (JustinRMoy) [inline resources/group.py:29]: Why would students be able to create classes? This sounds like something only a professor should be able to do

**Review by JustinRMoy — COMMENTED** [review]
This commit specifically should be undone entirely. These should remain as Posts, you should have seen when I made this change from Get to Post when you did my code review

```
[Bugfixes for some get requests] 
[5aaf4df]
Namely GetMentorQuestions and GetMultipleChoiceOptions
```

File: utils.py (hunk 3 above)
Comment (Keilvin) [reply utils.py]: I think here it's just easier to understand

File: resources/group.py (hunk 4 above)
Comment (Keilvin) [reply resources/group.py:29]: ST stands for Student, PF stands for Professor and SU stands for Super User

File: resources/mentors.py (hunk 2 above)
Comment (Keilvin) [reply resources/mentors.py:323]: Got it

File: resources/group.py (hunk 4 above)
Comment (JustinRMoy) [reply resources/group.py:29]: I see

File: utils.py (hunk 3 above)
Comment (JustinRMoy) [reply utils.py]: It's  a pointless change and unnecessarily creates a new variable while not really making a real change as the values are still assigned the same way but not with extra steps

**Review by JustinRMoy — CHANGES_REQUESTED** [review]
address my last comment and it should be good to go

File: utils.py (hunk 3 above)
Comment (Keilvin) [reply utils.py]: ok

File: utils.py [hunk 5]
```diff
@@ -403,11 +403,14 @@ def getTimeDiffFormatted(time_1 = None,
...
         return None, None
 
+    if isinstance(time_delta, str):
+        return None, None
+
     if time_delta.days != 0:
         d = {"days": time_delta.days}
         d["hours"], rem = divmod(time_delta.seconds, 3600)
         d["minutes"], d["seconds"] = divmod(rem, 60)
-        if d['days'] != 1 and d['days'] != -1:
```
Comment (JustinRMoy) [inline utils.py:402]: why the change from AND to OR?

**Review by JustinRMoy — CHANGES_REQUESTED** [review]

**Review by JustinRMoy — APPROVED** [review]
